### PR TITLE
Tupha/support card level select action uwp

### PR DESCRIPTION
**Pull request by tuanh118**
Card-level `selectAction` works as specified in the **Proposed Solution** from https://github.com/Microsoft/AdaptiveCards/issues/1199

### CODE REVIEW

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 1]
```diff
@@ -153,6 +153,14 @@ AdaptiveNamespaceStart
             THROW_IF_FAILED(MakeAndInitialize<AdaptiveRenderArgs>(&bodyRenderArgs, containerStyle, childElementContainerAsFE.Get()));
             BuildPanelChildren(body.Get(), bodyElementContainer.Get(), renderContext, bodyRenderArgs.Get(), [](IUIElement*) {});
 
+            ComPtr<IAdaptiveActionElement> selectAction;
+            THROW_IF_FAILED(adaptiveCard->get_SelectAction(&selectAction));
+
+            // Create a new IUIElement pointer to house the root element decorated with select action
+            ComPtr<IUIElement> rootSelectActionElement;
+            HandleSelectAction(nullptr, selectAction.Get(), renderContext, rootElement.Get(), SupportsInteractivity(hostConfig.Get()), true, &rootSelectActionElement);
+            THROW_IF_FAILED(rootSelectActionElement.As(&childElementContainerAsFE));
```
Comment (tuanh118) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: test codeflow

Comment (tuanh118) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: test resolved comment

---
In reply to: [183123853](https://github.com/Microsoft/AdaptiveCards/pull/1396#discussion_r183123853) [](ancestors = 183123853)

File: source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl [hunk 2]
```diff
@@ -576,6 +576,9 @@ AdaptiveNamespaceStart
         [propget] HRESULT Style([out, retval] ContainerStyle* value);
         [propput] HRESULT Style([in] ContainerStyle value);
 
+		[propget] HRESULT SelectAction([out, retval] IAdaptiveActionElement** value);
```
Comment (RebeccaAnne) [inline source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl]: >		 [](start = 0, length = 2)

Change tabs to spaces please

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 3]
```diff
@@ -2597,27 +2605,32 @@ AdaptiveNamespaceStart
...
         // We don't know what the spacing of the NEXT element will be, so we can't calculate the correct spacing below.
         // For now, we'll simply assume the bottom spacing is the same as the top.
-        ABI::AdaptiveNamespace::Spacing elementSpacing;
-        THROW_IF_FAILED(adaptiveCardElement->get_Spacing(&elementSpacing));
-        UINT spacingSize;
-        THROW_IF_FAILED(GetSpacingSizeFromSpacing(hostConfig.Get(), elementSpacing, &spacingSize));
-        double topBottomPadding = spacingSize / 2.0;
+        // NOTE: Only apply spacings (padding, margin) for adaptive card elements to avoid adding
+        // spacings to card-level selectAction.
+        if (adaptiveCardElement != nullptr) {
```
Comment (khouzam) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: > { [](start = 43, length = 2)

nit: brackets should be on the other line

Comment (tuanh118) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: Sorry JavaScript broke me. Fixed in 8bb22d0

File: source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl (hunk 2 above)
Comment (tuanh118) [reply source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl]: 🤦‍♂️  .Fixed in 8bb22d0

**Comment by BenAffleckIsBatman** [comment]
Hey there. The fact that UWP Renderer doesn't support selectAction is really missing for our implementation, that's why I looked for this merge. I forked the current master branch and created a nuget package out of it. However the selectAction is still not working in UWP renderer. Am I missing something? this merge should already closed the gap?

**Comment by tuanh118** [comment]
@BenAffleckIsBatman can you open a new issue with your app's details and we can continue the conversation there?

**Comment by BenAffleckIsBatman** [comment]
@tuanh118 sure! Did so #1572.

**Pull request merged**